### PR TITLE
Add git commit details to stress environment

### LINKED ISSUE #7521: Add env var with git commit hash to stress test build
I'd like to have a recorded git commit hash of the code being tested in my stress tests.
Having something like `--build-arg GIT_COMMIT=$(git rev-parse HEAD)` in the stress test deployment script would allow anyone to use this as they see please (e.g. stamp on telemetry for troubleshooting/reporting later)

EDIT (@benbp): Also add job/pod labels for git commit sha, so that queries can use the git commit to locate pods running it.

### PR DESCRIPTION
**Pull request by benbp**
Fixes #7521 #4644. See README.md diff in PR for details.

FYI @lmolkova @richardpark-msft

### CODE REVIEW

File: eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1 [hunk 1]
```diff
@@ -321,7 +321,20 @@ function DeployStressPackage(
...
+        "helm", $subCommand, $releaseName, $pkg.Directory,
+        "-n", $pkg.Namespace,
+        $subCommandFlag,
+        "--values", $generatedConfigPath,
+        "--set", "stress-test-addons.env=$environment"
+    )
+
+    Push-Location $pkg.Directory
+    $gitCommit = git rev-parse HEAD 2>&1
+    Pop-Location
```
Comment (weshaggard) [inline eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1:334]:
```suggestion
    $gitCommit = git --git-dir $pkg.Directory rev-parse HEAD 2>&1
```

**Review by weshaggard — APPROVED** [review]
One suggestion but otherwise seems good.

File: eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1 (hunk 1 above)
Comment (benbp) [reply eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1:334]: Ooh didn't know about that!

**Comment by benbp** [comment]
> One suggestion but otherwise seems good.

@weshaggard looks like `--git-dir` only works if pointing at the `.git` directory, but `-C` has the intended effect.

File: eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1 [hunk 2]
```diff
@@ -321,7 +321,18 @@ function DeployStressPackage(
...
-    $helmCommandArg = "helm", $subCommand, $releaseName, $pkg.Directory, "-n", $pkg.Namespace, $subCommandFlag, "--values", $generatedConfigPath, "--set", "stress-test-addons.env=$environment"
+    $helmCommandArg = @(
+        "helm", $subCommand, $releaseName, $pkg.Directory,
+        "-n", $pkg.Namespace,
+        $subCommandFlag,
+        "--values", $generatedConfigPath,
+        "--set", "stress-test-addons.env=$environment"
+    )
+
+    $gitCommit = git -C $pkg.Directory rev-parse HEAD 2>&1
```
Comment (weshaggard) [inline eng/common/scripts/stress-testing/stress-test-deployment-lib.ps1:332]: I don't think you want `-C` here do you? When I tried that locally I don't think it worked. I thought that was referring to the configuration location.

**Comment by weshaggard** [comment]
> > One suggestion but otherwise seems good.
> 
> @weshaggard looks like `--git-dir` only works if pointing at the `.git` directory, but `-C` has the intended effect.

That is interesting as that I had the opposite when I tested locally. I could point to the folder that contained the `.git` directory and I thought `-C` was for the configuration file.

**Pull request merged**